nemo/nemo/utils/exp_logging.py

```diff
@@ -80,20 +83,43 @@ def __init__(
         # tb_dir behaves the same as ckpt_dir except default folder name is
         # tensorboard instead of checkpoints
         self.tb_dir = tb_dir
+        tm_suf = time.strftime('%Y-%m-%d_%H-%M-%S')
+        if local_rank is not None and add_time:
+            if broadcast_func is None:
+                raise ValueError(
+                    "local rank was not None, but ExpManager was not passed a "
+                    "broadcast function to broadcast the datetime suffix")
+            if local_rank == 0:
+                broadcast_func(string=tm_suf)
+            else:
+                tm_suf = broadcast_func(str_len=len(tm_suf))
 
         # Create work_dir if specified
         if work_dir:
```

Comment:
I am strongly voting for standardizing the output dir to: "~/experiments" (or sth similar) and adding that as "default variable".
Two reasons:
1) some examples are polluting and writing whatever they need to ".", this fixes it once and for all
2) standardizing will ease the use, as the user will *always* know where to search your results

Reply:
I would argue against this. If you are running on a cloud instance, you would want to specify the exact path that you want to save to. The current default in NeuralFactory is to not create any logging or work directories. Users can overwrite this however they want in their own scripts.

Reply:
1) " If you are running on a cloud instance, you would want to specify the exact path that you want to save to. "

I do not see that as a showstopper for adding *default output dir* to "~/experiments", am I missing something here?

2) "The current default in NeuralFactory is to not create any logging or work directories."

Any reason for NOT logging results from a given experiment?

3) "Users can overwrite this however they want in their own scripts."

Exactly. That is why we are talking about the *default output dir*.

Reply:
I would be ok with default being creating a logdir. But why a default output dir? Why not just "."?

Reply:
I feel like this is now beyond the scope of this PR. The point of this PR was to fix 2 bugs within ExpManager.

Reply:
I already explained why not ".'

And if the scope of PR is "Clean up ExpManager's logging directory creation" then how this could be beyond?

Reply:
PR is now rename to "Bugfix For ExpManager" which is what I am trying to do

Reply:
Yeah I'm with Tom on this.  We should have reasonable defaults.  "." is a bad default.

Reply:
I would argue against ""~/experiments", to me "." is a more reasonable default:
(1) A lot of our use-cases are running using docker containers and running on cluster/cloud instances.  
(2) On another side of spectrum - beginner users, unfamiliar with NeMo, will not know where to look. Worse yet, if stuff is being saved, and they don't suspect it is then them restarting stuff will mean restoring checkpoints et.c

Reply:
I also prefer standardizing to a default directory, since "." has caused some clutter locally for me. This should be added to the contributing guidelines under NeMo Style once we decide on a default.

Re whether the change should be in this PR, I agree with Jason--if the scope here was to fix bugs, then similarly to the license headers, we should probably make a separate PR to change all the default dirs across all applicable files.

Reply:
My main issue is actually "~" vs "."  If it is "~/experiments" or "./experiments" <- I don't have strong opinion.
But if we do: "~/experiments"  The problem is what if the user already has "/home/username/experiments" with valuable data? We might be polluting it without the user realizing that. This is why I think "." or "./experiments" is better because then the user will notice that.
In either case, @jonathan-cohen-nvidia please make call on this one.

Reply:
@tkornuta-nvidia are we good to change that to "./experiments" and merge this PR?

Reply:
ping @tkornuta-nvidia " are we good to change that to "./experiments" and merge this PR?"